### PR TITLE
submitted transaction listener

### PR DESCRIPTION
**Pull request by cellog**
# Description

This is the first of 2 steps in the transaction handling flow. It is designed to be called declaratively for all transactions, regardless of prior state. If the newest transaction for the current key is in a later state (confirming, valid, etc.) then this will return immediately.

Otherwise, it waits for the `'transaction.new'` event from walletService and responds by updating the transaction and key state

Probably the final code will look something like:

```js
const submittedTransaction = Promise.all([
  purchaseKey({
    walletService,
    window,
    lockAddress,
    amountToSend
  }),
  submittedListener({
    lockAddress,
    existingTransactions,
    existingKeys,
    walletService,
    requiredConfirmations,
  })
])
// cache the submitted transaction
// use it as defaults for web3Service.getTransaction
```

# Issues

Refs #3206

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: paywall/src/data-iframe/blockchainHandler/keyStatus.js [hunk 1]
```diff
@@ -42,8 +42,7 @@ export function getKeyStatus(key, requiredConfirmations) {
 /**
  * Construct the transactions field for each key
  * @param {object} keys keys, indexed by their lock/owner ID (not the same as smart contract ID)
- * @param {object} transactions transactions, indexed by hash (submitted transaction is always
```
Comment (cellog) [inline paywall/src/data-iframe/blockchainHandler/keyStatus.js]: correct a misunderstanding. This is not directly related to this PR, but is indirectly related

File: paywall/src/data-iframe/blockchainHandler/purchaseKey/submittedListener.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    locks: [lockAddress],
+    requiredConfirmations,
+  })
+  const account = getAccount()
+  const network = getNetwork()
+  const keyToPurchase = `${lockAddress}-${account}`
+  const key = keys[keyToPurchase]
+  if (
+    (key.status !== 'none' && key.status !== 'expired') ||
+    key.status === 'failed'
```
Comment (julien51) [inline paywall/src/data-iframe/blockchainHandler/purchaseKey/submittedListener.js]: I really think we should constant-ize the key status here because I have to say I am very blurry on what are the "allowed" values for a status. What does 'failed' mean actually? Is that the transaction status? If so is that the latest transaction or another one?

Comment (cellog) [reply paywall/src/data-iframe/blockchainHandler/purchaseKey/submittedListener.js]: The key status is always linked to the most recent transaction.

Comment (cellog) [reply paywall/src/data-iframe/blockchainHandler/purchaseKey/submittedListener.js]: You can see how the key status is generated in `keyStatus.js` which is in master

Comment (cellog) [reply paywall/src/data-iframe/blockchainHandler/purchaseKey/submittedListener.js]: here: https://github.com/unlock-protocol/unlock/blob/master/paywall/src/data-iframe/blockchainHandler/keyStatus.js#L19

Comment (cellog) [reply paywall/src/data-iframe/blockchainHandler/purchaseKey/submittedListener.js]: I pushed a commit, it doesn't "constant-ize" the statuses, as we don't have direct control over some of the statuses that come from transaction status. It does fix the logic inconsistency: we should allow purchasing a new key when the previous transaction failed.

**Comment by cellog** [comment]
ready for a 2nd look @julien51
